File: 10253.py
import math


def main():
    t = int(input())
    for _ in range(t):
        a, b = (int(x) for x in input().split())
        while a > 1:
            x = b // a + 1
            a, b = a * x - b, b * x
            a, b = a // math.gcd(a, b), b // math.gcd(a, b)
        print(b)


main()

File: 10253.cpp
#include <cstdio>

int gcd(int a, int b){
	if(b == 0){
		return a;
	}else{
		return gcd(b, a % b);
	}
}

int main(){
	int t;
	scanf("%d", &t);
	for(int i = 0; i < t; ++i){
		int a, b;
		scanf("%d %d", &a, &b);
		while(a > 1){
			int x = b / a + 1;
			a = a * x - b;
			b = b * x;
			int g = gcd(a, b);
			a /= g;
			b /= g;
		}
		printf("%d\n", b);
	}
	return 0;
}
